src/main/java/com/sonymobile/jenkins/plugins/mq/mqnotifier/RunListenerImpl.java

```diff
@@ -61,67 +63,74 @@ public RunListenerImpl() {
 
     @Override
     public void onStarted(Run r, TaskListener listener) {
-        if (r instanceof AbstractBuild) {
-            AbstractBuild<?, ?> build = (AbstractBuild<?, ?>)r;
-            List<String> causes = new LinkedList<String>();
-            for (Object o : build.getCauses()) {
-                causes.add(o.getClass().getSimpleName());
-            }
+        List<String> causes = new LinkedList<String>();
+        for (Object o : r.getCauses()) {
+            causes.add(o.getClass().getSimpleName());
+        }
 
-            Cause.UpstreamCause upstreamCause = (Cause.UpstreamCause)r.getCause(Cause.UpstreamCause.class);
-            if (upstreamCause != null) {
-                causes.add(upstreamCause.getShortDescription());
+        Cause.UpstreamCause upstreamCause = (Cause.UpstreamCause)r.getCause(Cause.UpstreamCause.class);
+        if (upstreamCause != null) {
+            causes.add(upstreamCause.getShortDescription());
 
-                TopLevelItem item = Jenkins.getInstance().getItem(upstreamCause.getUpstreamProject());
-                if (item != null && item instanceof MatrixProject) {
-                    //Find the build
-                    MatrixBuild mb = ((MatrixProject)item).getBuildByNumber(upstreamCause.getUpstreamBuild());
-                    causes.add(mb.getUrl());
-                }
-            }
-            Cause.RemoteCause remoteCause = (Cause.RemoteCause)r.getCause(Cause.RemoteCause.class);
-            if (remoteCause != null) {
-                causes.add(remoteCause.getShortDescription());
-            }
-            Cause.UserIdCause userIdCause = (Cause.UserIdCause)r.getCause(Cause.UserIdCause.class);
-            if (userIdCause != null) {
-                causes.add(userIdCause.getShortDescription());
+            TopLevelItem item = Jenkins.getInstance().getItem(upstreamCause.getUpstreamProject());
+            if (item != null && item instanceof MatrixProject) {
+                //Find the build
+                MatrixBuild mb = ((MatrixProject)item).getBuildByNumber(upstreamCause.getUpstreamBuild());
+                causes.add(mb.getUrl());
             }
+        }
+        Cause.RemoteCause remoteCause = (Cause.RemoteCause)r.getCause(Cause.RemoteCause.class);
```

Comment:
Yes, apparently  the compiler does not know which class type getCause() will return.